### PR TITLE
Add new profiles to hadge

### PR DESCRIPTION
**Pull request by wxicu**
1. add profiles `conda`, `docker`, `arm`, `test`, like in nf-core
2. Change some values of parameter from `FALSE` to `False`, reduce mixed-case problem
3. Specify conda configuration in conda directive. Customized conda configuration is recommended to specify in a new profile now
4. docs is adapted

### CODE REVIEW

**Comment by wxicu** [comment]
@Zethson ready for review

File: test.config [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  common_variants_souporcell =  "$projectDir/test_data/common_variants_hg19.vcf"
+  common_variants_freemuxlet = "$projectDir/test_data/jurkat_293t_exons_only.vcf.withAF.vcf.gz"
+  common_variants_cellsnp  = "$projectDir/test_data/genome1K.phase3.SNP_AF5e2.chr1toX.hg19.vcf.gz"
+  vcf_donor = "$projectDir/test_data/jurkat_293t_exons_only.vcf.withAF.vcf"
+  // Call freebayes on chr 1 and chr 2 only to speed up run time
+  region = "1;2"
+  // donor genotype file provided by popscle doesnt work on souporcell
+  use_known_genotype = "False"
+  ignore = "True"
+
```
Comment (Zethson) [inline test.config:26]:
```suggestion
```

File: test.config [hunk 2]
```diff
@@ -0,0 +1,27 @@
+params {
+  outdir = "result_test"
+  // input for hashing-based deconvolution
+  hto_matrix_raw = "$projectDir/test_data/hto"
```
Comment (Zethson) [inline test.config:4]: Ideally all of these should eventually be fetched directly from the cloud but let's not bother with that for now. Downloading them with the script is OK.

File: modules/single/hash_demultiplexing.nf [hunk 3]
```diff
@@ -14,7 +14,9 @@ include { bff_hashing } from './hash_demulti/bff'
 process summary{
     publishDir "$projectDir/$params.outdir/$params.mode/hash_demulti", mode: 'copy'
     label 'small_mem'
-    label 'summary'
+
+    conda "python=3.9 'pandas<2.0' scanpy muon numpy"
```
Comment (Zethson) [inline modules/single/hash_demultiplexing.nf:18]:
```suggestion
    conda "python=3.9 'pandas<2.0' scanpy muon"
```

numpy should already be included in scanpy.

Is it a lot of work to make this pandas 2.x compatible? That'd be great together with increasing the Python version to 3.11

File: modules/single/hash_demulti/demuxem.nf [hunk 4]
```diff
@@ -5,6 +5,8 @@ process demuxem{
     publishDir "$projectDir/$params.outdir/$params.mode/hash_demulti/demuxem", mode:'copy'
     label 'small_mem'
     
+    conda "python=3.9 bioconda::pegasuspy 'pandas<2.0.0' demuxEM"
```
Comment (Zethson) [inline modules/single/hash_demulti/demuxem.nf:8]: Is it a lot of work to make this pandas 2.x compatible? That'd be great together with increasing the Python version to 3.11

**Review by Zethson — COMMENTED** [review]
This is really really awesome. What happens now if I specify Docker or Conda and attempt to run Souporcell? We're not gracefully handling the issue, are we?

 Now we should be able to bring it all together:

Please add a Github Actions workflow that runs the test profile using https://github.com/marketplace/actions/setup-nextflow
If everything is well designed and works with Conda/Docker (except for souporcell) this should hopefully be a piece of cake.

This will be a nice major milestone.

Thank you very much!

**Comment by wxicu** [comment]
> This is really really awesome. What happens now if I specify Docker or Conda and attempt to run Souporcell? We're not gracefully handling the issue, are we?
> 
> Now we should be able to bring it all together:
> 
> Please add a Github Actions workflow that runs the test profile using https://github.com/marketplace/actions/setup-nextflow If everything is well designed and works with Conda/Docker (except for souporcell) this should hopefully be a piece of cake.
> 
> This will be a nice major milestone.
> 
> Thank you very much!

Is it possible to upload the test data to cloud which can be used by github workflow action directly? Reference genome is tooooo large...

**Comment by Zethson** [comment]
Ohhh I see. nf-core has small test reference genomes. See: https://github.com/nf-core/rnaseq/blob/master/conf/test.config#L27

**Pull request merged**